src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py

```diff
@@ -142,17 +148,16 @@ def flexible_server_restore(cmd, client,
     else:
         source_server_id = source_server
 
-    try:
-        restore_point_in_time = datetime.strptime(restore_point_in_time, "%Y-%m-%dT%H:%M:%S.%f+00:00")
-    except ValueError:
-        restore_point_in_time = datetime.strptime(restore_point_in_time, "%Y-%m-%dT%H:%M:%S+00:00")
+    restore_point_in_time = change_str_to_datetime(restore_point_in_time)
     restore_point_in_time = restore_point_in_time.replace(tzinfo=dt.timezone.utc)
 
     parameters = postgresql_flexibleservers.models.Server(
         point_in_time_utc=restore_point_in_time,
         source_server_name=source_server,  # this should be the source server name, not id
         create_mode="PointInTimeRestore",
         availability_zone=zone,
+        source_resource_group_name=resource_group_name,
+        source_subscription_id=get_subscription_id(cmd.cli_ctx),
```

Comment:
`get_subscription_id(cmd.cli_ctx)` is getting the login subscription. Is that possible that the `resource_group_name` or `source_server` belong to other sub?

Reply:
It won't be the case. It looks like there's complicate story from the backend team but we concluded that we keep the restore behavior as it was (no source server from different rg nor different sub)